definitions/npm/styled-components_v4.x.x/flow_v0.104.x-/styled-components_v4.x.x.js

```diff
@@ -245,44 +251,60 @@ declare module 'styled-components' {
 
   declare export function withTheme<Theme, Config: {...}, Instance>(Component: React$AbstractComponent<Config, Instance>): React$AbstractComponent<$Diff<Config, ThemeProps<Theme | void>>, Instance>
 
-  declare export type StyledComponent<Props, Theme, Instance> = React$AbstractComponent<Props, Instance> & InterpolatableComponent<Props>
-  declare interface StyledFactory<StyleProps, Props, Theme, Instance> {
+  // NOTE: I removed interpolable component because it breaks jsx instantiation
```

Comment:
Forgot to delete this comment?

Reply:
Yeah, inference of props doesn't work either way no matter how I tried

Reply:
Yeah, Flow's a bit fickle here. It seems to infer props sometimes, but sometimes it just fails.